spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1016,6 +1016,10 @@ def _create_client_for_kernel(self, connection_file, hostname, sshkey,
             cf_path = cf_path if cf_path else None
             connection_file = find_connection_file(filename=cf_filename,
                                                    path=cf_path)
+            if os.path.splitext(connection_file)[1] != ".json":
+                # There might be a file with the same id in the path.
+                connection_file = find_connection_file(
+                    filename=cf_filename + ".json", path=cf_path)
```

Comment:
Why is this necessary for this PR? Or better yet, could you explain why you're adding these lines?

Reply:
So when testing, the stderr file (and in the future, the .stdout and .fault) are returned by `find_connection_file`. This is because this code was moved from the plugin folder to the widget folder I think.

Reply:
The idea is that find_connection_file searches for a file with name matching the kernel ID. unfortunately the stderr file matches this so can be returned instead of the json file.